runtime/gc_realtime/MetronomeAlarm.cpp

```diff
@@ -207,29 +207,22 @@ MM_ITAlarm::initialize(MM_EnvironmentBase *env, MM_MetronomeAlarmThread* alarmTh
 	}
 	
 #if defined(WIN32)
-	TIMECAPS tc;
-	if (timeGetDevCaps(&tc, sizeof(TIMECAPS)) != TIMERR_NOERROR) {
-		return false;
-	}
-	UINT uDesiredPeriod = (UINT)(_extensions->itPeriodMicro / 1000);
-	UINT uPeriod = (uDesiredPeriod < tc.wPeriodMin) ? tc.wPeriodMin : uDesiredPeriod;
-	UINT uDelay = uPeriod;
-	if (timeBeginPeriod(uPeriod) != TIMERR_NOERROR) {
-		/* Error beginning higher frequency period  */
-		return false;
-	}
-	_uTimerId = (UINT)timeSetEvent(uDelay, uPeriod, itAlarmThunk, (DWORD_PTR)alarmThread, TIME_PERIODIC);
-	if (_uTimerId == 0) {
+	OMRPORT_ACCESS_FROM_ENVIRONMENT(env);
+	DWORD dwPeriod = (DWORD)(_extensions->itPeriodMicro / 1000);
+	DWORD dwDelay = dwPeriod;
+	BOOL ret = CreateTimerQueueTimer(&_hTimer, NULL, (WAITORTIMERCALLBACK)itAlarmThunk, (PVOID)alarmThread, dwDelay, dwPeriod, 0);
```

Comment:
You are assuming this order (...., PVOID Parameter, DWORD DueTime, DWORD Period, ULONG Flags) of parameters (which is not what MS docs suggests)?

Reply:
The docs are wrong. That same MS page links to an example which uses the function in this way instead. Note that on the doc page, it has PVOID DueTime and ULONG Parameter, which doesn't make sense.